lib/commons/color/get-background-stack.js

```diff
@@ -95,31 +82,89 @@ function sortPageBackground(elmStack) {
 }
 
 /**
- * Get all elements rendered underneath the current element,
- * In the order they are displayed (front to back)
- *
- * @method getBackgroundStack
- * @memberof axe.commons.color
- * @param {Element} elm
- * @return {Array}
+ * Determine if the element obscures / overlaps with the text
+ * @private
+ * @param {Number} elmIndex
+ * @param {Array} elmStack
+ * @param {Element} originalElm
+ * @param {Object} options
+ * @return {Number|undefined}
  */
-function getBackgroundStack(elm) {
-  let elmStack = filteredRectStack(elm);
-
-  if (elmStack === null) {
-    return null;
+function calculateObscuringElement(elmIndex, elmStack, originalElm, options) {
+  // Reverse order, so that we can safely splice
+  for (let i = elmIndex - 1; i >= 0; i--) {
+    if (
+      !isInlineDescendant(originalElm, elmStack[i]) &&
+      !isEmptyElement(elmStack[i], options)
+    ) {
+      return true;
+    }
+    // Ignore inline descendants, for example:
+    // <p>text <img></p>; We don't care about the <img> element,
+    // since it does not overlap the text inside of <p>
+    elmStack.splice(i, 1);
   }
-  elmStack = reduceToElementsBelowFloating(elmStack, elm);
-  elmStack = sortPageBackground(elmStack);
+  return false;
+}
 
-  // Return all elements BELOW the current element, null if the element is undefined
-  const elmIndex = elmStack.indexOf(elm);
-  if (calculateObscuringElement(elmIndex, elmStack, elm)) {
-    // if the total of the elements above our element results in total obscuring, return null
-    incompleteData.set('bgColor', 'bgOverlap');
-    return null;
+/**
+ * Determine if element B is an inline descendant of A
+ * @private
+ * @param {Element} node
+ * @param {Element} descendant
+ * @return {Boolean}
+ */
+function isInlineDescendant(node, descendant) {
+  const CONTAINED_BY = Node.DOCUMENT_POSITION_CONTAINED_BY;
+  // eslint-disable-next-line no-bitwise
+  if (!(node.compareDocumentPosition(descendant) & CONTAINED_BY)) {
+    return false;
   }
-  return elmIndex !== -1 ? elmStack : null;
+  const style = window.getComputedStyle(descendant);
+  const display = style.getPropertyValue('display');
+  if (!display.includes('inline')) {
+    return false;
+  }
+  // IE needs this; It doesn't set display:block when position is set
+  const position = style.getPropertyValue('position');
+  return position === 'static';
 }
 
-export default getBackgroundStack;
+/**
+ * Determine if an element is empty and would not contribute to the background color
+ * @see https://github.com/dequelabs/axe-core/issues/3464
+ * @private
+ * @param {Element} node
+ * @param {Object} options
+ */
+function isEmptyElement(node, { pseudoSizeThreshold, ignorePseudo }) {
+  const vNode = getNodeFromTree(node);
+  const style = window.getComputedStyle(node);
+  const bgColor = getOwnBackgroundColor(style);
+  const hasPseudoElement = !!findPseudoElement(vNode, {
```

Comment:
This method doesn't seem right here. For example, if the pseudo element isn't position:absolute, `hasPsuedoElement` would become false, even if there's a background color, or content, or a bunch of other things on that.

I'm not sure how to fix that, except perhaps maybe have this be "true" unless `content:none` is true for both `:before` and `:after`? I'm not sure that `findPseudoElement` should have been added to `commons`. There are a bunch of things it does that make sense for inside color-contrast-evaluate, but that create problematic gotchas if used in other places.